core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -703,8 +703,8 @@ class KafkaServer(
                 case None =>
                   info(s"Broker registration for controller $controllerId is not available in the metadata cache")
               }
-            case None =>
-              info("No controller present in the metadata cache")
+            case Some(_: KRaftCachedControllerId) | None =>
```

Comment:
Hmm, looks like you replaced an exhaustive pattern matching block with a non exhaustive one? Am I missing something?

Reply:
getControllerId returns Option(CachedControllerId). CachedControllerId has two inherited classes: 1) ZkCachedControllerId and 2) KRaftCachedControllerId.

So we need to cover 1) None, 2) Some(KRaftCachedControllerId), and 3) Some(ZkCachedControllerId). I'm covering all these conditions. Did I miss something, or are you saying we might miss some future inheritance cases?

BTW, I had the below code before. But that seems to have made it confusing to the reviewers. Let me know if that works for you @ijuma 

```
getControllerId match {
  case Some(controllerId: ZkCachedController) => ...
  case _ => ...
```

Reply:
The snippet in your comment ^ looks good to me @akhileshchg. In the previous PR, my comment was just about avoiding the filter + `isInstanceOf`. The `Some(controllerId: ZkCachedController)` covers that

Reply:
@akhileshchg My bad, I hadn't seen the `| None` part.

Reply:
I am not sure why a revert commit was added after this discussion. `case _` disables exhaustiveness checks and hence should be used sparingly.

Reply:
I thought @mumrah was suggesting using `case _`. Let me revert it again then. Also, why would `case _` disable the exhaustiveness check, isn't it like `default`?

Reply:
Okay. Removed the change and reverted it to explicit match-case.